lib/azure_blob_store.rb

```diff
@@ -79,6 +79,15 @@ def cdn_url(url)
       url.sub("#{schema}#{absolute_base_url}", SiteSetting.azure_cdn_url)
     end
 
+    def url_for(upload, force_download: false)
+      if force_download
+        uri = URI.parse(upload.url)
+        uri.query = [uri.query, "rscd=file;%20attachment"].compact.join('&')
```

Comment:
I'm not sure if it's really that easy to set the `Content-Disposition` header. You might need something like [`SharedAccessSignature#signed_uri`](https://github.com/Azure/azure-storage-ruby/blob/master/common/lib/azure/storage/common/core/auth/shared_access_signature_generator.rb) to generate a signed URL.

Reply:
According to [the Azure docs](https://docs.microsoft.com/en-us/rest/api/storageservices/constructing-a-service-sas#specifying-query-parameters-to-override-response-headers-blob-service-and-file-service-only), this is how to set the `Content-Disposition` header. I'm seeing the correct header in the response that the browser receives:
![Screen Shot 2019-08-08 at 2 31 40 PM](https://user-images.githubusercontent.com/516388/62703427-7bf9d580-b9e9-11e9-95d9-b145f3a5429e.png)

Reply:
Actually... maybe that's not the correct header... let me investigate a little more.

Reply:
I'm pretty sure the `rscd=file;%20attachment` gets ignored, because as per documentation it needs to be part of a signed URL. But the plugin already sets the `Content-Disposition` header in https://github.com/discourse/discourse-azure-blob-storage/blob/22fcf8d84dc503183c9d98575e29be57b179f5e7/lib/azure_blob_store.rb#L21
That's why you see the correct header. You might want to update that line to use `ActionDispatch::Http::ContentDisposition.format(disposition: "attachment", filename: filename)` though.

So, you shouldn't need any special handling for `force_download`. At least not until the plugin starts supporting private uploads like the [`S3Store` does](https://github.com/discourse/discourse/blob/3b8c4688326641f66e0f1737ffd770c6e325874a/lib/file_store/s3_store.rb#L114).

Reply:
You're right @gschlager 

Although I do think [there is a case for handling `force_download` for downloading lightboxed images](https://github.com/discourse/discourse/pull/7845).

For now I've simplified `url_for` to fix the original bug, but I'm going to look into how to set the `Content-Disposition` header at the time of the request, rather than just at the time of upload.